src/core.js

```diff
@@ -2082,6 +2098,7 @@ define([
 
     fn.handleMugRename = function (form, mug, newId, oldId, newPath, oldPath, oldParent) {
         form.handleMugRename(mug, newId, oldId, newPath, oldPath, oldParent);
+        window.location.hash = '#' + mug.hashtagPath;
```

Comment:
Does `mug.hashtagPath` not start with `#`? At any rate, see suggestion above about dropping the extra `#`.